### PR TITLE
Add an integration test for calling clearPersistence() on a new Firestore instance

### PR DESCRIPTION
**Pull request by dconeybe**

### CODE REVIEW

File: packages/firestore/test/integration/api/database.test.ts [hunk 1]
```diff
@@ -1096,6 +1096,30 @@ apiDescribe('Database', (persistence: boolean) => {
...
+        const firestore = docRef.firestore;
+        await docRef.set({ foo: 'bar' });
+        const app = docRef.firestore.app;
+        const name = app.name;
+        const options = app.options;
+
+        await app.delete();
+        const app2 = firebase.initializeApp(options, name);
+        const firestore2 = firebase.firestore!(app2);
+        await firestore2.clearPersistence();
```
Comment (dconeybe) [inline packages/firestore/test/integration/api/database.test.ts:1113]: Note that the _only_ difference between this test case and the one above it is that the location of the call to clearPersistence() has moved to be invoked on the newly-created `firestore2` object.

Comment (schmidt-sebastian) [reply packages/firestore/test/integration/api/database.test.ts:1112]: Just checking - is this test meaningfully different from:

```
// eslint-disable-next-line no-restricted-properties
(persistence ? it : it.skip)( 'can clear persistence if the client has not been initialized',
  () => withTestDb(db => db.clearPersistence())
)
```

You should also update your PR title. You are writing a super fancy integration test, and not a unit test :)

Comment (dconeybe) [reply packages/firestore/test/integration/api/database.test.ts:1112]: The "difference" between that 1-line test is that it doesn't verify the side effects of `clearPersistence()`. Unfortunately, the extra lines all exist to provide the verification that `clearPersistence()` does indeed clear the persistence. Also, I will update the PR description as suggested.

Comment (schmidt-sebastian) [reply packages/firestore/test/integration/api/database.test.ts:1112]: Makes sense. I did not quite catch that the first time around.

**Pull request merged**